### DISCUSSION TITLE
MQTT Problem

### DISCUSSION THREAD

**Discussion by Dengo91** [syssi/homeassistant-goecharger-mqtt#161] [General]
Hi, i try to get in running but it will not work so fare. I used the following steps but i can´t see data like in step 6 described. is there something that is not described?

1) Use HACS to install this custom component
2) Restart HA
3) Make sure you have an MQTT Broker ([e.g. Mosquitto broker](https://github.com/eclipse/mosquitto)) and a working [MQTT integration](https://www.home-assistant.io/integrations/mqtt/) in HA
4) (Recommended) Access to messages on MQTT Broker via client such as [MQTT Explorer](https://mqtt-explorer.com/)
5) Enable the MQTT API of your charger at the [go-eCharger App](https://play.google.com/store/apps/details?id=co.goe.app&hl=de)
Internet -> Advanced settings -> scroll to bottom -> Enable MQTT API ([See also here](https://github.com/goecharger/go-eCharger-API-v2/blob/main/mqtt-en.md))
at the field URL enter your MQTT broker URL, e.g. mqtt://192.168.178.3:1883 for a broker running on a server with IP 192.168.178.3 on default port 1883 alt text
(optional) Add MQTT keys and certificates as needed
Press the SAVE symbol in the top right to apply the settings
6) (Optional) Make sure you see a new topic on the MQTT broker using your MQTT client
New messages appear under go-eCharger/[your serial number]/
7) In HA, add new integration
Settings -> Devices & services -> ADD INTEGRATION -> go-eCharger (MQTT)
enter serial number (6 digits) of your device (find via App or MQTT client)
make sure base topic matches what you see in the MQTT broker, in some cases the default /go-eCharger has to be changed to go-eCharger
8) See your go-e data pour into HA

**Comment by syssi** [comment]
Could you post a screenshot of the incoming data? I would like to double check the topic. Does it start with a slash or not? Does it include the serial number?

**Reply by Dengo91** [reply]
![MQTT](https://github.com/syssi/homeassistant-goecharger-mqtt/assets/171798362/1e56e635-8a2d-420c-946d-45b36d4a13a6)

Hey, it´s not showing the go data. :(

**Reply by syssi** [reply]
Could you provide another screenshot of the MQTT settings at the Android app? It looks like the device cannot talk to the MQTT server at the moment. Does your MQTT password include any special chars?

**Reply by Dengo91** [reply]
![WhatsApp Bild 2024-06-07 um 09 47 00_2ff0de1c](https://github.com/syssi/homeassistant-goecharger-mqtt/assets/171798362/e4700057-ec4b-4d89-ab09-a3bf113d0976)
 you mean this?

**Reply by syssi** [reply]
Yes. Is your MQTT broker (192.168.178.88) able to accept connection without a username & password? If not the URL should look like this:

```
mqtt://username:password@192.168.178.88:1883
```

**Reply by Dengo91** [reply]
oh yes that was the problem :). Thank you for your support

**Reply by Dengo91** [reply]
ok i have another question:). How i can use the Charge with PV Surplus. I have a sensor for my pv produktion.

**Reply by syssi** [reply]
PV Surplus charging is a really big and complex topic depending on your setup. Please read as much as possible and talk to the community. 

These are the most active discussions I'm aware of:

https://github.com/syssi/homeassistant-goecharger-mqtt/issues/101
https://community.home-assistant.io/t/custom-component-goecharger-mqtt-go-echarger-homefix-ev-charger-integration/394420

**Comment by JosefinBerlin** [comment]
Hallo Syssi,
ich bin ein völliger newbie und scheitere ab Punkt 7 der Installation. MQTT Server läuft, in MQTT Explorer sehe ich, dass Daten kommen.
![MQTT Explorer](https://github.com/user-attachments/assets/0c2569ba-f956-48f3-83e1-855f34ae15bb)
Wenn ich nun in Geräte & Dienste eine Integration hinzufügen möchte, sehe ich go-eCharger (MQTT) nicht.
Was mache ich falsch?

**Reply by syssi** [reply]
Hast du die Komponente per HACS installiert?

**Reply by JosefinBerlin** [reply]
uups, das hatte ich übersehen. Bin gleich zu Geräte und Dienste. Jetzt ist es heruntergeladen und installiert. Es läuft. Danke für die Hilfe.
